Docs/details/santad.md

```diff
@@ -0,0 +1,21 @@
+# santad
+
+The santad process does the heavy lifting when it comes to making decisions about binary executions. It also handles brokering all of the XPC connections between the various components of Santa. It does all of this with performance being at the forefront. 
+
+##### A note on performance
+
+On an idling machine santad, and really all of the components of Santa, consumes virtually no CPU and a minimal amount of memory (5-50MB). When lots of processes `exec()` at the same time, the CPU and memory usage can spike. All of the `exec()` decisions are made on a high priority threads to ensure decisions are posted back to the kernel as soon as possible. There is a watchdog thread that will log warnings when there is sustained high CPU (>20%) and memory (>250MB) usage by santad.
+
+##### On Launch
+
+The very first thing santad does once it has been launched is to load and connect to santa-driver. Only one connection may be active at any given time.
+
+At this point santa-driver is loaded and running in the kernel but is allowing all executions and not sending any messages to santad. Before santad tells santa-driver it is ready to receive messages it needs to setup a few more things.
```

Comment:
At this point, santa-driver is loaded and running in the kernel, but is allowing all executions and not sending any messages to santad. Before santad tells santa-driver it is ready to receive messages, it needs to setup a few more things.